warehouse/transformer/internal/utils/utils_test.go

```diff
@@ -87,26 +92,86 @@ func TestValidTimestamp(t *testing.T) {
 	}
 }
 
-// BenchmarkValidTimestamp/ValidTimestamp_Valid
-// BenchmarkValidTimestamp/ValidTimestamp_Valid-12         				36466681		32.00 ns/op
-// BenchmarkValidTimestamp/ValidTimestamp_Invalid
-// BenchmarkValidTimestamp/ValidTimestamp_Invalid-12       	 			2823615	       	423.4 ns/op
-// BenchmarkValidTimestamp/ValidTimestamp_Invalid_Big_String
-// BenchmarkValidTimestamp/ValidTimestamp_Invalid_Big_String-12     	7731496	       	154.8 ns/op
+// BenchmarkValidTimestamp
+// BenchmarkValidTimestamp/old_parser
+// BenchmarkValidTimestamp/old_parser-12         	    6999	    163017 ns/op
+// BenchmarkValidTimestamp/new_parser
+// BenchmarkValidTimestamp/new_parser-12         	   89427	     13392 ns/op
 func BenchmarkValidTimestamp(b *testing.B) {
-	b.Run("ValidTimestamp_Valid", func(b *testing.B) {
-		for i := 0; i < b.N; i++ {
-			ValidTimestamp("2023-11-10T12:34:56Z")
+	timeFormats := []string{
+		time.RFC3339Nano,
+		time.DateOnly,
+		misc.RFC3339Milli,
+		time.RFC3339,
+		time.RFC1123Z,
+		time.RFC1123,
+		time.RFC822Z,
+		time.RFC822,
+		time.UnixDate,
+		time.DateTime,
+		time.RubyDate,
+		time.ANSIC,
+		"2006-01-02 15:04:05.999999999 -0700 MST",
+		"2006-01-02T15:04:05.999999999",
+		"2006-01-02T15:04:05",
+		"2006-01-02 15:04:05.999999999",
+		"2006-01-02 15:04:05",
+		"2006-01-02T15:04:05",
+		"2006-01-02 15:04:05.000-0700",
+	}
+
+	testDates := []string{
+		"2012/03/19 10:11:59",
+		"2012/03/19 10:11:59.3186369",
+		"2009-08-12T22:15:09-07:00",
+		"2014-04-26 17:24:37.3186369",
+		"2012-08-03 18:31:59.257000000",
+		"2013-04-01 22:43:22",
+		"2014-04-26 17:24:37.123",
+		"2014-12-16 06:20:00 UTC",
+		"1384216367189",
+		"1332151919",
+		"2014-05-11 08:20:13,787",
+		"2014-04-26 05:24:37 PM",
+		"2014-04-26",
+	}
+	for i := 0; i < 100; i++ {
+		testDates = append(testDates, strings.Repeat("a", 1000))
+	}
+	for i := 0; i < 100; i++ {
+		testDates = append(testDates, uuid.NewString())
+	}
+
+	oldValidTimestamp := func(input string) bool {
+		if len(input) > validTimestampFormatsMaxLength {
+			return false
 		}
-	})
-	b.Run("ValidTimestamp_Invalid", func(b *testing.B) {
+		var t time.Time
+		var err error
+
+		for _, format := range timeFormats {
+			t, err = time.Parse(format, input)
+			if err == nil {
+				break
+			}
+		}
+		return !t.IsZero() && !t.Before(minTimeInMs) && !t.After(maxTimeInMs)
+	}
+
+	b.Run("old parser", func(b *testing.B) {
```

Comment:
Should we get rid of old parser benchmark since we removed it from the code?

Reply:
For now, we need it for Benchmarking.